### PR TITLE
fix: host should not return a redirect source file from `getSourceFile`

### PR DESCRIPTION
**Pull request by alan-agius4**
## I'm submitting a...

```
[x] Bug Fix
[ ] Feature
[ ] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [x] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [x] Tests for the changes have been added

## Description
This fix is basicilly because of recursive node_modules when you do `npm link` or using `lerna`

https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping

closes: #473

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by alan-agius4** [comment]
@dherges can you have a look at this? This is super critical for me  :) Thanks

File: src/lib/conf/tsconfig.ngc.json [hunk 1]
```diff
@@ -24,7 +23,12 @@
     "lib": [
       "dom",
       "es2015"
-    ]
+    ],
+    "paths": {
+      "*": [
```
Comment (dherges) [inline src/lib/conf/tsconfig.ngc.json]: Why is that wild-card mapping?

**Comment by alan-agius4** [comment]
So it captures all modules and tried to resolve then in the main
node_modules instead. Afaik, the last time I check Angular CLI had the same
thing.

**Comment by dherges** [comment]
Ok, I am going to "cross-check" with the CLI code...

File: src/lib/conf/tsconfig.ngc.json (hunk 1 above)
Comment (dherges) [reply src/lib/conf/tsconfig.ngc.json]: Can you post me a link how CLI does implement this?

I just found this doc:; https://github.com/angular/angular-cli/blob/1156befb930faf6458040bb234d0b46b11dff21e/docs/documentation/stories/linked-library.md

Comment (alan-agius4) [reply src/lib/conf/tsconfig.ngc.json]: Actually I cannot find this any more, Though what I found is a test  https://github.com/angular/angular-cli/blob/master/tests/e2e/tests/build/ts-paths.ts#L31

This works slightly different, but same concept. Try to resolve the module provided using the standard resolution and if not resolved use `app/shared/*'`

**Comment by devoto13** [comment]
This looks like a hacky way to disable optimization introduced by https://github.com/Microsoft/TypeScript/pull/16274 :)

Angular CLI has same issue - https://github.com/angular/angular-cli/issues/8445#issuecomment-360323029. It looks like bug/limitation of TypeScript.

**Comment by alan-agius4** [comment]
This might not be needed with RC 13 and new Javascript API. I'll have a look when the new RC is out.

**Comment by alan-agius4** [comment]
Can be solved by using ng-packagr api

**Comment by gmfun** [comment]
@alan-agius4 how can this be solved using ng-packagr api

**Comment by alan-agius4** [comment]
@gmfun are you using a mono repo by any chance? i solved mine by providing a tsconfig with paths for build.

I have a skeleton repo with an example of this.
https://github.com/alan-agius4/ng-mono-repo-starter/blob/master/tsconfig.build.json
https://github.com/alan-agius4/ng-mono-repo-starter/blob/master/tools/tasks/build.ts


If you can provide an example of your problem maybe i can help you out a bit better.

**Comment by gmfun** [comment]
@alan-agius4 https://github.com/hypertrack/js-sdk this the repo where I am using angular with lerna (packages/angular-components). Currently stuck at typescript 2.4 as upgrading gives the error. Will also look into the repo you have shared and try to understand how you fixed, it. I first glance it looks like you are using ng-packagr api to override tsconfig.

**Comment by alan-agius4** [comment]
Yes, indeed I am using the API to override the tsconfig.

Ill try to have a look at your repo in the coming days, as at the moment I
am on a short trip.

**Comment by alan-agius4** [comment]
When i try to run `build` under `angular-components` I am getting a bunch of error like;

```bash

BUILD ERROR
src/app/pipes/time-string.pipe.ts(2,26): error TS2307: Cannot find module 'ht-utility'.
src/app/pipes/date-string.pipe.ts(3,26): error TS2307: Cannot find module 'ht-utility'.
src/app/pipes/name-case.pipe.ts(2,24): error TS2307: Cannot find module 'ht-utility'.
src/app/pipes/dot.pipe.ts(2,25): error TS2307: Cannot find module 'ht-utility'.
src/app/pipes/date-humanize.pipe.ts(2,28): error TS2307: Cannot find module 'ht-utility'.
src/app/pipes/distance-locale.pipe.ts(2,30): error TS2307: Cannot find module 'ht-utility'.
src/app/pipes/hm-string.pipe.ts(2,24): error TS2307: Cannot find module 'ht-utility'.
src/app/ht/ht-map.service.ts(2,37): error TS2307: Cannot find module 'ht-maps'.
src/app/users-container/users-container.component.ts(8,21): error TS2307: Cannot find module 'ht-ut
....
...
```

**Pull request closed without merging**